include/fmt/format.h

```diff
@@ -2925,9 +2937,25 @@ struct formatter<T, Char,
           &specs_, internal::char_specs_checker<decltype(eh)>(specs_.type, eh));
       break;
     case internal::type::float_type:
+#if FMT_USE_FLOAT
+      internal::parse_float_type_spec(specs_, eh);
+#else
+      FMT_ASSERT(false, "float support disabled");
+#endif
+      break;
     case internal::type::double_type:
+#if FMT_USE_DOUBLE
+      internal::parse_float_type_spec(specs_, eh);
+#else
+      FMT_ASSERT(false, "double support disabled");
+#endif
+      break;
     case internal::type::long_double_type:
+#if FMT_USE_LONG_DOUBLE
       internal::parse_float_type_spec(specs_, eh);
+#else
+      FMT_ASSERT(false, "long double support disabled");
+#endif
```

Comment:
Conditional compilation is not really needed here, just use normal `if` statements.

Reply:
Done.